src/muya/lib/contentState/searchCtrl.js

```diff
@@ -34,11 +34,23 @@ const matchString = (text, value, options) => {
 }
 
 const searchCtrl = ContentState => {
+  ContentState.prototype.buildRegexValue = function (match, value) {
+    const groups = value.match(/\$(\d)*/g)
+    if (groups) {
+      for (const regexGroup of groups) {
+        const groupIndex = regexGroup[1] - 1
+        if (groupIndex < match.subMatches.length) {
```

Comment:
`groupIndex` can be `-1` if you use `$0`.

```suggestion
        if (groupIndex >= 0 && groupIndex < match.subMatches.length) {
```